master/internal/rm/agentrm/resource_pool.go

```diff
@@ -484,13 +484,13 @@ func (rp *resourcePool) Receive(ctx *actor.Context) error {
 		ctx.Respond(resourceSummaryFromAgentStates(rp.agentStatesCache))
 
 	case sproto.CapacityCheck:
+		reschedule = false
 		var totalSlots int
 		switch {
-		case rp.config.Provider == nil:
-			rp.agentStatesCache = rp.fetchAgentStates(ctx)
-			defer func() {
-				rp.agentStatesCache = nil
-			}()
+		case rp.config.Provider == nil || rp.config.Provider.HPC != nil:
```

Comment:
```rp.config.Provider.HPC``` was added I'm assuming making the panic go off in some cases? 

Not sure how this option works if this is valid to use the same agent counting logic.